thanos/charts/templates/servicemonitor.yaml

```
@@ -0,0 +1,21 @@
+# SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Greenhouse contributors
+# SPDX-License-Identifier: Apache-2.0
+
+{{ if .Values.thanos.serviceMonitorLabels }}
```

Cool. But I would suggest 
```suggestion
{{ if .Values.thanos.selfMonitor.enabled }}
```

---

I see your point, I would go for how kube-prometheus-stack does it then
  
serviceMonitor:
    ## If true, create a serviceMonitor for prometheus operator
    ##
    selfMonitor: true

    ## Labels for ServiceMonitor
    additionalLabels: {}